android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/navigation/NavGraph.kt

```diff
@@ -4,33 +4,50 @@ import androidx.compose.runtime.Composable
 import androidx.navigation.NavController
 import androidx.navigation.NavGraphBuilder
 import androidx.navigation.compose.NavHost
-import androidx.navigation.compose.composable
 import androidx.navigation.compose.rememberNavController
-import androidx.navigation.compose.navigation
-import co.nimblehq.ic.kmm.suv.android.ui.screens.login.LoginScreen
 import co.nimblehq.ic.kmm.suv.android.ui.screens.home.HomeScreen
+import co.nimblehq.ic.kmm.suv.android.ui.screens.home.SurveyArgument
+import co.nimblehq.ic.kmm.suv.android.ui.screens.login.LoginScreen
+import co.nimblehq.ic.kmm.suv.android.ui.screens.surveydetail.SurveyDetailScreen
 
 @Composable
 fun NavGraph() {
     val navController = rememberNavController()
-    NavHost(navController = navController, startDestination = Route.AUTHENTICATION) {
+    NavHost(navController = navController, startDestination = AppDestination.Authentication.route) {
         authenticationGraph(navController)
         mainGraph(navController)
     }
 }
 
 fun NavGraphBuilder.authenticationGraph(navController: NavController) {
-    navigation(startDestination = Route.LOGIN, route = Route.AUTHENTICATION) {
-        composable(Route.LOGIN) {
+    navigation(
+        startDestination = AppDestination.Login,
+        destination = AppDestination.Authentication
+    ) {
+        composable(AppDestination.Login) {
             LoginScreen(onLogInSuccess = {
-                navController.navigate(Route.HOME)
+                navController.navigate(AppDestination.Home)
             })
         }
     }
 }
 
 fun NavGraphBuilder.mainGraph(navController: NavController) {
-    composable(Route.HOME) {
-        HomeScreen()
+    composable(AppDestination.Home) {
+        HomeScreen(onSurveyDetailClick = {
+            navController.currentBackStackEntry?.arguments?.putParcelable(Argument.survey, it)
+            navController.navigate(AppDestination.Survey)
+        })
+    }
+
+    composable(AppDestination.Survey) {
+        val arguments = navController.previousBackStackEntry?.arguments
+        val survey = arguments?.getParcelable<SurveyArgument>(Argument.survey)
+        SurveyDetailScreen(
+            survey,
+            onBackClick = {
+                navController.popBackStack()
```

Comment:
we should use navigateUp here https://stackoverflow.com/questions/67245601/difference-between-navigateup-and-popbackstack

```suggestion
                navController.navigateUp()
```

Reply:
Fixed in https://github.com/suho/kmm-ic/pull/112/commits/7fbd237f360d78de95ab0e58b0ae313e0de32918